spec.html

```diff
@@ -21514,8 +21462,19 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
         1. Set the running execution context's LexicalEnvironment to _classScope_.
         1. Set the running execution context's PrivateEnvironment to _classPrivateEnvironment_.
         1. If _constructor_ is ~empty~, then
-          1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-default-constructor-functions" title></emu-xref>.
-          1. Let _F_ be ! CreateBuiltinFunction(_steps_, 0, _className_, &laquo; [[ConstructorKind]], [[SourceText]] &raquo;, the current Realm Record, _constructorParent_).
+          1. Let _defaultConstructor_ be a new Abstract Closure with no parameters that captures nothing and performs the following steps when called:
+            1. Let _args_ be the List of arguments that was passed to this function by [[Call]] or [[Construct]].
+            1. If NewTarget is *undefined*, throw a *TypeError* exception.
+            1. Let _F_ be the active function object.
+            1. If _F_.[[ConstructorKind]] is ~derived~, then
+              1. NOTE: This branch behaves similarly to `constructor(...args) { super(...args); }`. The most notable distinction is that while the aforementioned ECMAScript source text observably calls the @@iterator method on `%Array.prototype%`, this function does not.
+              1. Let _func_ be ! _F_.[[GetPrototypeOf]]().
+              1. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
+              1. Return ? Construct(_func_, _args_, NewTarget).
+            1. Else,
+              1. NOTE: This branch behaves similarly to `constructor() {}`.
+              1. Return ? OrdinaryCreateFromConstructor(NewTarget, *"%Object.prototype%"*).
+          1. Let _F_ be ! CreateBuiltinFunction(_defaultConstructor_, 0, _className_, &laquo; [[ConstructorKind]], [[SourceText]] &raquo;, the current Realm Record, _constructorParent_).
```

Comment:
```suggestion
          1. Let _F_ be ! CreateBuiltinFunction(_defaultConstructor_, 0, _className_, &laquo; [[ConstructorKind]] &raquo;, the current Realm Record, _constructorParent_).
```

Reply:
This PR doesn't interact with the `[[SourceText]]` slot at all. It should remain. (It's used to [store the source of the class itself](https://tc39.es/ecma262/multipage/ecmascript-language-functions-and-classes.html#sec-runtime-semantics-bindingclassdeclarationevaluation).)